### DISCUSSION TITLE
How does sharing work?

### DISCUSSION THREAD

**Discussion by thorrapp** [cloud-py-api/nc_py_api#262] [Q&A]
I've got the following code.  Uploading the file works properly, but sharing gives me a Typerror when calling update.  

I'm assuming I have to run Update after creating the share, as trying to remove the update function call runs successfully, but when I go to the web interface there's no share created, and trying to print the URL via print(share.url()) gives me a Recursion error (maximum recursion depth exceeded).

> http://127.0.0.1:8000/Mittelstands-Magazin/versand/
> 5.0.4
> TypeError
> list indices must be integers or slices, not str
> C:\Users\tr6\AppData\Local\Programs\Python\Python311\Lib\site-packages\nc_py_api\files\__init__.py, line 381, in share_id
> magazine.views.versand
> C:\Users\tr6\AppData\Local\Programs\Python\Python311\python.exe
> 3.11.8

Code Snippet:
```
    upload = nc.files.upload_stream('magazine/' + os.path.basename(up.uploaddatei), up.uploaddatei)
    print(str(upload))
    share = nc.files.sharing.create(upload, share_type=3, permissions=1, password="temp", expire_date=dt.today() + relativedelta(weeks=+1))
    nc.files.sharing.update(share)
    print(share.path())
```

**Comment by bigcat88** [comment]
```python3

if __name__ == "__main__":
    from datetime import datetime, timedelta
    import nc_py_api

    nc = nc_py_api.Nextcloud(nc_auth_user="admin", nc_auth_pass="admin", nextcloud_url="http://stable29.local")
    with open(__file__) as fp:
        upload = nc.files.upload_stream('dev3.py', fp)
    print(str(upload))
    share = nc.files.sharing.create(
        upload, share_type=3, permissions=1, password="temp", expire_date=datetime.now() + timedelta(weeks=1)
    )
    print(share.url)
```


Output:
```
File: `dev3.py` with id=00000336ocsa4sm3ydei last modified at 1970-01-01 00:00:00 and  permissions.
http://stable29.local/index.php/s/5k4Am2jpLBbcZwN
```    


`nc.files.sharing.update(share)` is only needed if you want to `update` some parameters of the share, and it must have at least one argument indicating what you want to change on the share.

**Reply by spawn-guy** [reply]
@bigcat88 will you be so kind to provide a "reverse operation" example?
i have an url `http://stable29.local/index.php/s/5k4Am2jpLBbcZwN` - how do i list and download the files from that share?
